packages/medusa/src/api-v2/admin/regions/[id]/route.ts

```diff
@@ -17,3 +22,38 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
 
   res.status(200).json({ region })
 }
+
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  const { result, errors } = await updateRegionsWorkflow(req.scope).run({
+    input: {
+      selector: { id: req.params.id },
+      update: req.validatedBody as UpdatableRegionFields,
+    },
+    throwOnError: false,
+  })
+
+  if (Array.isArray(errors) && errors[0]) {
```

Comment:
Is there a middleware that catches these errors and handles them gracefully? Wouldn't doing something like `res.status(500)` be better if not?

Reply:
we have a middleware that handle that, and we generally throw a medusa error type and depending on the type it respondwith the appropriate status